### ISSUE TITLE
Vignette Errors

### ISSUE THREAD

**Issue by schifferl** [vobencha/curatedMetagenomicData#8]
In order for the vignette to build, it was necessary to remove these lines (approximate location of 163 in curatedMetagenomicData_vignette.Rnw). I cannot seem to spot the error in the markup, perhaps you can @lwaldron or @vobencha.  

```
The results can be narrowed down by filtering by the body site 
``hard_palate":
<<example4loadHmpiistep2>>=
listResources("curatedMetagenomicData", c("hmp", "hard_palate"))
@
```

Additionally, it was necessary to comment out the argument on line # 198 curatedMetagenomicData_vignette.Rnw. Also, would need some help here deciphering the source of the error.

Otherwise, the vignette builds and allows for R CMD check to pass in its current state.

**Comment by vobencha** [comment]
The first problem is the underscore in hard_palate - it needs to be escaped, e.g., hard\_palate.

I'm not sure why the second error is occurring. Maybe the restructured search in list/loadResources() picks up different files than when Levi first wrote this. When debugging heatmap.2() there is a problem applying the distance function to 'x':
```
Browse[2]> 
debug: distr <- distfun(x)
Warning message:
In cor(t(c)) : the standard deviation is zero
```

This is what 'x' looks like:
```
Browse[2]> x[1:3, 1:3]
                                 H10      H11      H12
k__Archaea                   0.24169  0.50621  0.30522
k__Archaea|p__Euryarchaeota  0.24169  0.50621  0.30522
k__Bacteria                 99.75831 99.49379 99.69478
```

The error is thrown from hclustfun():
```
debug: hcr <- hclustfun(distr)
Browse[2]> 
Error in hclustfun(distr) : NA/NaN/Inf in foreign function call (arg 11)
```

Once the distfun issue is sorted out we need to find the missing curatedMetagenomicData:::.shortenCladeName() function defined in the 'labRow' arg. This may have been in the LeviRmisc package?

@schifferl do you have permissions to commit to the svn repo yet?
https://hedgehog.fhcrc.org/bioc-data/trunk/experiment/pkgs/curatedMetagenomicData

**Comment by lwaldron** [comment]
I just realized that I committed a new vignette to my own fork and didn't submit a pull request... The vignette in the pull request has no errors. There are probably some parts of the current vignette worth conserving and merging into the new Rmd one, as the new one focuses on single-dataset analysis, but perhaps start with the new one and add to it.

**Comment by lwaldron** [comment]
Actually, I think the only thing really missing from the new Rmd vignette is an example of selecting based on ExperimentHub metadata. The example in the original Rnw vignette uses the pData, so would require downloading all datasets before selecting datasets/subjects based on rules.

**Comment by schifferl** [comment]
I have merged the pull request with 25853c9 and will go through the Rmd vignette. I left the old Rnw vignette in place but will remove it once I move anything important to the Rmd file.

Otherwise, @vobencha still no access to the SVN source. Is Dan or Martin the right person to contact?

And the `shortenCladeName()` function did come from LeviRmisc package, I simply added it as a hidden function because I believe it is only used in the vignette.

So to close the issue I am going to move anything important to Rmd, remove the Rnw and bib files, and push the changes.

**Comment by schifferl** [comment]
Errors in the Rnw are corrected with d1818f6. The issue with the `distfun` argument was resolved by removing NA values from the matrix via a helper function.

I find that the Rnw vignette still contains useful information so I have actually decided not to remove it yet and will work on merging it with the Rmd version. A PDF of the built Rnw vignette is attached.

[curatedMetagenomicData_vignette.pdf](https://github.com/vobencha/curatedMetagenomicData/files/352981/curatedMetagenomicData_vignette.pdf)
